plugins/inputs/snmp2/snmp.go

```diff
@@ -0,0 +1,788 @@
+package snmp2
+
+import (
+	"bytes"
+	"fmt"
+	"math"
+	"net"
+	"os/exec"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+
+	"github.com/soniah/gosnmp"
+)
+
+const description = `Retrieves SNMP values from remote agents`
+const sampleConfig = `
+  agents = [ "127.0.0.1:161" ]
+  version = 2
+
+  # SNMPv1 & SNMPv2 parameters
+  community = "public"
+
+  # SNMPv2 & SNMPv3 parameters
+  max_repetitions = 50
+
+  # SNMPv3 parameters
+  #sec_name = "myuser"
+  #auth_protocol = "md5"         # Values: "MD5", "SHA", ""
+  #auth_password = "password123"
+  #sec_level = "authNoPriv"      # Values: "noAuthNoPriv", "authNoPriv", "authPriv"
+  #context_name = ""
+  #priv_protocol = ""            # Values: "DES", "AES", ""
+  #priv_password = ""
+
+  # measurement name
+  name = "system"
+  [[inputs.snmp2.field]]
+    name = "hostname"
+    oid = ".1.0.0.1.1"
+  [[inputs.snmp2.field]]
+    name = "uptime"
+    oid = ".1.0.0.1.2"
+  [[inputs.snmp2.field]]
+    name = "load"
+    oid = ".1.0.0.1.3"
+  [[inputs.snmp2.field]]
+    oid = "HOST-RESOURCES-MIB::hrMemorySize"
+
+  [[inputs.snmp2.table]]
+    # measurement name
+    name = "remote_servers"
+    inherit_tags = [ "hostname" ]
+    [[inputs.snmp2.table.field]]
+      name = "server"
+      oid = ".1.0.0.0.1.0"
+      is_tag = true
+    [[inputs.snmp2.table.field]]
+      name = "connections"
+      oid = ".1.0.0.0.1.1"
+    [[inputs.snmp2.table.field]]
+      name = "latency"
+      oid = ".1.0.0.0.1.2"
+
+  [[inputs.snmp2.table]]
+    # auto populate table's fields using MIB
+    oid = "HOST-RESOURCES-MIB::hrNetworkTable"
+`
+
+// execCommand is so tests can mock out exec.Command usage.
+var execCommand = exec.Command
+
+// execCmd executes the specified command, returning the STDOUT content.
+// If command exits with error status, the output is captured into the returned error.
+func execCmd(arg0 string, args ...string) ([]byte, error) {
+	out, err := execCommand(arg0, args...).Output()
+	if err != nil {
+		if err, ok := err.(*exec.ExitError); ok {
+			return nil, NestedError{
+				Err:       err,
+				NestedErr: fmt.Errorf("%s", bytes.TrimRight(err.Stderr, "\n")),
+			}
+		}
+		return nil, err
+	}
+	return out, nil
+}
+
+// Snmp2 holds the configuration for the plugin.
+type Snmp2 struct {
+	// The SNMP agent to query. Format is ADDR[:PORT] (e.g. 1.2.3.4:161).
+	Agents []string
+	// Timeout to wait for a response. Value is anything accepted by time.ParseDuration().
+	Timeout string
```

Comment:
the Timeout parameter should be in the output of SampleConfig. Also use an `internal.Duration` type, which calls time.ParseDuration at the time that the toml is parsed.